### PR TITLE
docs: update spec documentation and add events 

### PR DESCRIPTION
**Pull request by kogisin**
## Description

closes: #127 

## Tasks

- [x] Add events on spec docs `06_events.md`
- [x] Update spec docs (add more description of the message fields, add more contexts)
- [x] Update `docs/How-To/cli/README.md` to have context about `PrivatePlanFarmingPoolAddress` 
---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Appropriate labels applied
- [ ] Targeted PR against correct branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: x/farming/spec/06_events.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+| ----------------- | -------------------- | ------------------------ |
+| plan_terminated   | plan_id              | {planID}                 |
+| plan_terminated   | farming_pool_address | {farmingPoolAddress}     |
+| plan_terminated   | termination_address  | {terminationAddress}     |
+| rewards_allocated | plan_id              | {planID}                 |
+| rewards_allocated | amount               | {totalAllocatedAmount}   |
+
+## Handlers
+
+### CreateFixedAmountPlan
```
[the remark this answers is not in the document]
Comment (kogisin) [reply x/farming/spec/06_events.md:19]: Oops! Yes it should be `MsgCreateFixedAmountPlan `

File: x/farming/spec/06_events.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+| stake   | staking_coins | {stakingCoins}  | 
+| message | module        | farming         |
+| message | action        | stake           |
+| message | sender        | {senderAddress} |
+
+### MsgUnstake
+
+| Type    | Attribute Key   | Attribute Value  |
+| ------- | --------------- | ---------------- |
+| unstake | plan_id         | {planID}         |
```
[the remark this answers is not in the document]
Comment (kogisin) [reply x/farming/spec/06_events.md:61]: You're right. I'll remove this._

File: x/farming/spec/04_messages.md [hunk 3]
```diff
@@ -6,60 +6,63 @@ Messages (Msg) are objects that trigger state transitions. Msgs are wrapped in t
...
+	StakingCoinWeights sdk.DecCoins // staking coin weights for the plan
+	StartTime          time.Time    // start time of the plan
+	EndTime            time.Time    // end time of the plan
+	EpochAmount        sdk.Coins    // distributing amount for every epoch
 }
 ```
 
 ## MsgCreateRatioPlan
 
+This is one of the private plan type messages that anyone can create. A ratio plan plans to distribute amount of coins by ratio defined in `EpochRatio`. All coins that the creator has in balances are used. Internally, `PrivatePlanFarmingPoolAddress` is created and the creator should query the plan's farming pool address and send amount of coins to the farming pool address so that the plan distributes as intended. Note that there is a fee `PlanCreationFee` paid upon plan creation to prevent from spamming attack.
```
[the remark this answers is not in the document]
Comment (kogisin) [reply x/farming/spec/04_messages.md:24]: Remove the line because the fees move to newly created `FarmingPoolAddress` and whichever the creator sends to the farming pool address are used. Let me rephrase the sentence.

**Comment by kogisin** [comment]
While updating spec documentation, I come to realize `docs/How-To-/cli/README.md` needs to be updated as well. Let me make the updates.

**Review by hallazzang — APPROVED** [review]

**Review by dongsam — APPROVED** [review]
